addons/io_hubs_addon/components/components_registry.py

```diff
@@ -125,6 +176,12 @@ def load_components_registry():
                 register_component(member)
                 __components_registry[member.get_name()] = member
 
+    # Preferences are not accessible until the add-on is enabled so we need to check that before loading the user
+    # components as they depend on the preferences.
+    from ..utils import is_addon_enabled
+    if is_addon_enabled():
+        load_user_components()
```

Comment:
In which cases is the add-on not enabled enough when trying to load user components?  Is this related to running tests?
Also, I think this only gets called once, so if this check fails then I think the user components would never get loaded?

Reply:
Yes, when running tests the add-on is added but not enabled so the access to preferences fails so that's the case this is trying to fix.

When the add-on is disabled preferences ar lost anyway so if the add-on is not enabled you will need to add the user paths and restart anyways.